ddtrace/tracer/sqlcomment.go

```diff
@@ -188,7 +190,53 @@ func commentQuery(query string, tags map[string]string) string {
 	return b.String()
 }
 
-// Extract is not implemented on SQLCommentCarrier
+// Extract parses for key value attributes in a sql query's comment in order to build a span context
 func (c *SQLCommentCarrier) Extract() (ddtrace.SpanContext, error) {
-	return nil, nil
+	var ctx spanContext
+	re := regexp.MustCompile(`/\*(.*?)\*/`) // extract sql comment
+	if match := re.FindStringSubmatch(c.Query); len(match) == 2 {
+		comment := match[1]
+		kvs := strings.Split(comment, ",")
+		for _, unparsedKV := range kvs {
+			if splitKV := strings.Split(unparsedKV, "="); len(splitKV) == 2 {
+				key := splitKV[0]
+				value := strings.Trim(splitKV[1], "'")
+				switch key {
```

Comment:
We also potentially parse the parent service, DB service, parent version, and env. I wasn't sure if / where it belonged in the span context

Reply:
I'll let the tracing experts chime in but I think what you're doing is correct by ignoring those. They're applicable to Database monitoring when we need some basic info about the caller and aren't always guaranteed to be able to find it via the trace data.